src/modules/hip/kernel/to_decibels.hpp

```diff
@@ -0,0 +1,313 @@
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);
+    to_decibels_hip_compute(&src_f8, &dst_f8, minRatio, multiplier, inverseMagnitude);
+
+    uint dstIdx = (id_z * dstStride) + id_x;
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+}
+
+__global__ void to_decibels_2d_hip_tensor(float *srcPtr,
+                                          uint2 srcStridesNH,
+                                          float *dstPtr,
+                                          uint2 dstStridesNH,
+                                          RpptImagePatchPtr srcDims,
+                                          double minRatio,
+                                          float multiplier,
+                                          float *inverseMagnitudeTensor)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= srcDims[id_z].width || id_y >= srcDims[id_z].height)
+        return;
+
+    uint srcIdx = (id_z * srcStridesNH.x) + (id_y * srcStridesNH.y) + id_x;
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x;
+    float inverseMagnitude = inverseMagnitudeTensor[id_z];
+    dstPtr[dstIdx] = multiplier * log2(max(minRatio, (static_cast<double>(srcPtr[srcIdx]) * inverseMagnitude)));
+}
+
+// -------------------- Set 2 -  kernels for finding inverse magnitude value --------------------
+
+__global__ void inverse_magnitude_hip_tensor(float *srcPtr,
+                                             int maxLength,
+                                             bool computeMax,
+                                             float *inverseMagnitudeTensor)
+
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    // Do final reduction on block wise max
+    if (computeMax)
+    {
+        uint srcIdx = id_z * maxLength;
+        __shared__ float max_smem[256];
+        max_smem[hipThreadIdx_x] = srcPtr[srcIdx];
+
+        if (id_x >= maxLength)
+            return;
+
+        srcIdx += id_x;
+        float maxVal = srcPtr[srcIdx];
+        while (id_x < maxLength)
+        {
+            maxVal = fmaxf(maxVal, srcPtr[srcIdx]);
+            id_x += hipBlockDim_x;
+            srcIdx += hipBlockDim_x;
+        }
+        max_smem[hipThreadIdx_x] = maxVal;
+        __syncthreads();
+
+        // do reduction on min_smem and max_smem
+        for (int threadMax = 128; threadMax >= 1; threadMax /= 2)
+        {
+            if (hipThreadIdx_x < threadMax)
+                max_smem[hipThreadIdx_x] = max(max_smem[hipThreadIdx_x], max_smem[hipThreadIdx_x + threadMax]);
+            __syncthreads();
+        }
+
+        if (hipThreadIdx_x == 0)
+            inverseMagnitudeTensor[id_z] = 1.f / max_smem[0];
+    }
+    else
+    {
+        inverseMagnitudeTensor[id_z] = 1.0f;
```

Comment:
Pls check performance again without this if condition. 
Suggest using a handle allocated pinned memory preset of size say 1024 * sizeof(float) to 1.0f, 1.0f... at handle init itself. Do we support a batch size > 1024/2048?

Reply:
Yes it is supported
Currently there is no restriction on batchsize

Reply:
we cannot use handle memory here since because of issues that can happen with batch processing. For example
for batch1 computeMax is set to true, then we will update the handle memory
for batch2 if computeMax is set to false, then we will not update the handle memory use it as is (single we assume the memory is to 1.0f)

This will cause issues

I tried with hipMemset and removed if condition, but it is not working. Because hipMemcpy will set each byte to 1.0f which when for 4 bytes will become value > 1.0f

So the QA tests are failing. I am retaining the code as is for now for this case alone. Please let me know if there is any other way you want me to try